HelloSignalR/Startup.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Microsoft.AspNet.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.AspNet.Hosting;
+using HelloSignalR.Connections;
+using Microsoft.Extensions.Logging;
+using HelloSignalR.Providers;
+using Microsoft.AspNet.Http;
+using System;
+using Microsoft.AspNet.Authentication.JwtBearer;
+using System.Threading.Tasks;
+
+namespace HelloSignalR
+{
+    public class Startup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddAuthentication();
+            services.AddSignalR();
+            services.AddCaching();
+        }
+
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory, ILogger<Startup> logger)
+        {
+            // Add the platform handler to the request pipeline.
+            app.UseIISPlatformHandler();
+
+            if (env.IsDevelopment())
+            {
+                // In case any errors occur.
+                app.UseDeveloperExceptionPage();
+
+                if (env.IsDevelopment())
+                {
+                    loggerFactory.MinimumLevel = LogLevel.Information;
+                    loggerFactory.AddConsole();
+                    loggerFactory.AddDebug();
+                }
+            }
+
+            logger.LogInformation($"Environment: {env.EnvironmentName}");
+
+            var hostname = "http://localhost:5000/";
+
+            // Add token issuing middleware.
+            app.UseOpenIdConnectServer(config =>
+            {
+                // Your Auhentication server.
+                config.Provider = new AuthenticationProvider();
+                config.Issuer = new Uri(hostname);
+
+                if (env.IsDevelopment())
+                {
+                    // For not requiring HTTPS in localhost.
+                    config.AllowInsecureHttp = true;
+                    config.ApplicationCanDisplayErrors = true;
+                }
+            });
+
+            app.UseJwtBearerAuthentication(options =>
+            {
+                // We need this to enable authentication for all requests.
+                options.AutomaticAuthenticate = true;
+
+                // Your audience that has to be allowed by your Authority server.
+                options.Audience = hostname;
+
+                // Authority that issued your token (i.e. your identity server).
+                options.Authority = hostname;
+
+                options.Events = new JwtBearerEvents
+                {
+                    // For SignalR connections default Authorization header does not work,
+                    // Because there are no Headers in WebSockets specification
```

Comment:
Well, WS specification does not require Headers implementation, which is why WS JS API does not implement it. Why .NET client implements that is another question.